.github/workflows/main.yaml

```diff
@@ -0,0 +1,307 @@
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ matrix.SOURCE_IMAGE_REGISTRY }}
+        username: ${{ secrets[matrix.REGISTRY_USERNAME_SECRET_NAME]}}
+        password: ${{ secrets[matrix.REGISTRY_PASSWORD_SECRET_NAME]}}
+     
+    - name: Create Docker file
+      if: contains(fromJSON('["aries-endorser-proxy"]'), matrix.service)
+      run: |
+        BASE_IMAGE="${SOURCE_IMAGE_REGISTRY}${SOURCE_IMAGE_NAME}:${SOURCE_IMAGE_TAG}"
+        echo "$BASE_IMAGE"
+        mkdir ${context} && cd ${context} 
+        echo "FROM ${BASE_IMAGE}" > Dockerfile
+        echo "RUN chown 1001:root /usr/bin/caddy" >> Dockerfile
+      env:
+        context: ${{ matrix.SOURCE_CONTEXT_DIR }}
+        SOURCE_IMAGE_REGISTRY: ${{ matrix.SOURCE_IMAGE_REGISTRY }}
+        SOURCE_IMAGE_NAME: ${{ matrix.SOURCE_IMAGE_NAME }}
+        SOURCE_IMAGE_TAG: ${{ matrix.SOURCE_IMAGE_TAG }}
+
+    - name: Prepare docker tags for image
+      id: meta
+      if: contains(fromJSON('["aries-endorser-db","aries-endorser-agent","aries-endorser-backup","aries-endorser-api","aries-endorser-proxy"]'), matrix.service)
+      uses: docker/metadata-action@v5
+      with:
+        images: ${{ env.GITHUB_IMAGE_REPO }}${{ matrix.service }}
+        flavor: |
+          latest=true
+        tags: |
+          type=schedule
+          type=ref,event=branch
+          type=ref,event=pr
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}}
+          type=sha,value=latest     
+        labels: | 
+          io.openshift.build.source-location=${{ github.repositoryUrl }}
+          io.openshift.build.commit.id=${{ github.sha }}
+          
+    - name: Update Docker base image 
+      if: matrix.SOURCE_IMAGE_REGISTRY  != '' && contains(fromJSON('["aries-endorser-agent","aries-endorser-backup","aries-endorser-api","aries-endorser-proxy"]'), matrix.service)
+      run: |  
+        BASE_IMAGE="${SOURCE_IMAGE_REGISTRY}${SOURCE_IMAGE_NAME}:${SOURCE_IMAGE_TAG}"
+        sed -i -e "s;FROM .*;FROM ${BASE_IMAGE};g" "$file"
+      env:
+        context: ${{ matrix.SOURCE_CONTEXT_DIR }}
+        SOURCE_IMAGE_REGISTRY: ${{ matrix.SOURCE_IMAGE_REGISTRY }}
+        SOURCE_IMAGE_NAME: ${{ matrix.SOURCE_IMAGE_NAME }}
+        SOURCE_IMAGE_TAG: ${{ matrix.SOURCE_IMAGE_TAG }}
+        file: ${{ matrix.SOURCE_CONTEXT_DIR }}/${{ matrix.DOCKER_FILE_PATH }}
+
+    - name: Extract Tags
+      id: extract
+      if: contains(fromJSON('["aries-endorser-db"]'), matrix.service)
+      run: echo "tags=$(echo '${{  steps.meta.outputs.tags  }}' | grep -oE ':([^[:space:]]+)' |  sed '/workflow/d' | sed 's/://g' | tr '\n' ' ')" >> $GITHUB_OUTPUT
+
+    - name: Pull database image
+      if: contains(fromJSON('["aries-endorser-db"]'), matrix.service)
+      run: |
+        docker pull  ${{ matrix.SOURCE_IMAGE_REGISTRY }}${{ matrix.SOURCE_IMAGE_NAME }}:${{ matrix.SOURCE_IMAGE_TAG }}   
```

Comment:
The documentation for `redhat-actions/s2i-build` seems to indicate the docker image should not have to be pulled first.

Reply:
Will move this to an enhancement ticket.